### ISSUE TITLE
Check possibility of GEOlocation through integration with LF AssetManager

### ISSUE THREAD

**Issue by arruda** [openagri-eu/farmcalendar#11]
Agstack Assetmanager: https://github.com/agstack/asset-registry

**Comment by arruda** [comment]
Probably need to change hardcoded JWT alg:
https://github.com/agstack/asset-registry/blob/master/utils.py#L89 
and possibly remove the user activation requirement for now: https://github.com/agstack/asset-registry/blob/master/utils.py#L92

And also ignore the login/logout endpoints.

**Comment by arruda** [comment]
Using the asset-manager the geo locations in other places (ie. FarmCalendar) would only be geoID from the asset-registry.

**Comment by arruda** [comment]
@fedjo @nikoskal @pranavnbapat 
This is from the initial check I did on the integration with the Agstack asset registry.

As I mentioned before we would probably need to do a fork (for now, and later on a PR) of the asset manager to remove the hardcoded JWT token definition, so that we can use the same JWT algorithm from GK. And possibly remove the user activation requirement as well, or check it by default for the admin user (which is the user FC will be using).

And finally, we will then need to replace the UUID method with one that does a request in agstack when the model is saved:
https://github.com/openagri-eu/farmcalendar/blob/main/farm_management/models/base.py#L48
We might need to use the history details on that model so that we only call the agstack if the geometry field changed from the previous version that was saved on the DB, this way we don't call agstack on every save of the farm parcel, only on those that would lead to a different UUID.

**Comment by fedjo** [comment]
I agree with Felipe in the part of integration. After forking the repository in OpenAgri's repo, we need to modify the JWT authentication in asset-reg to match with OpenAgri ADS authentication system. Probably we need also to **_dockerize_** the service in order to be able to be deployed like other OA services.

**Comment by arruda** [comment]
@fedjo @nikoskal @pranavnbapat Just updating the latest decision:
We will go with the use of the AssetManager as an external service, similar to how we do with the weather data providers in the weather service. 
This means that when deploying the FC it should be possible to specify the URL of an asset-manager, and its credentials (or API keys/token, depending on how their system works on this). If this configurations are set, then the FC will use this service to (re)generate the parcel UUID whenever the geometry (WKT) changes (or when its first created).

if these env vars are not defined, then the FC reverts back to its local UUID5 generation (current behaviour)
